lib/postgres/postgres-pr/connection.rb

```diff
@@ -343,16 +347,34 @@ def handle_server_error_message(server_error_message)
 
   # tcp://localhost:5432
   # unix:/tmp/.s.PGSQL.5432
-  def establish_connection(uri, proxies)
+  def establish_connection(uri, proxies, ssl = nil)
     u = URI.parse(uri)
     case u.scheme
     when 'tcp'
       @conn = Rex::Socket.create(
-      'PeerHost' => (u.host || DEFAULT_HOST).gsub(/[\[\]]/, ''),  # Strip any brackets off (IPv6)
-      'PeerPort' => (u.port || DEFAULT_PORT),
-      'proto' => 'tcp',
-      'Proxies' => proxies
-    )
+        'PeerHost' => (u.host || DEFAULT_HOST).gsub(/\[|\]/, ''),
+        'PeerPort' => (u.port || DEFAULT_PORT),
+        'proto' => 'tcp',
+        'Proxies' => proxies
+      )
```

Comment:
To get the parameters to later be passed to `#initsock_with_ssl_version`, we'll need to convert them here ourselves.

```ruby
      params = Rex::Socket::Parameters.from_hash(
        'PeerHost' => (u.host || DEFAULT_HOST).gsub(/\[|\]/, ''),
        'PeerPort' => (u.port || DEFAULT_PORT),
        'proto' => 'tcp',
        'Proxies' => proxies,
        'SSLVersion' => ssl_version,
        'SSLVerifyMode' => ssl_verify_mode,
        'SSLKeyLogFile' =>
        'SSLCipher'...
      )
      @conn = Rex::Socket.create_param(params)
```